File: HelloWorld.cpp
#include <stdio.h>
#include <iostream>
using namespace std;

int main(void)
{
	string user_name;
	cout << "Enter your name: ";
	getline (cin,user_name);
	cout << "Hello world, " << user_name << "!" << endl;
	return 0;
}
// vim:sw=4: 

File: HelloWorld.java
import java.util.Scanner;
public class HelloWorld {
	public static void main (String[] args) {
		System.out.print("Enter your name: ");
		Scanner input = new Scanner(System.in);
		String user_name = input.next();
		System.out.println("Hello world, " + user_name + "!");
	}
}	
// vim:sw=4: 
